railties/lib/rails/command/environment_argument.rb

```diff
@@ -0,0 +1,32 @@
+require "active_support"
+
+module Rails
+  module Command
+    module EnvironmentArgument #:nodoc:
+      extend ActiveSupport::Concern
+
+      included do
+        argument :environment, optional: true, banner: "environment"
+      end
+
+      private
+        def extract_environment_option_from_argument
+          if environment
+            self.options = options.merge(environment: acceptable_environment(environment))
+          end
+        end
+
+        def acceptable_environment(env = nil)
+          if available_environments.include? env
+            env
+          else
+            %w(production development test).detect { |e| e =~ /^#{env}/ } || env
```

Comment:
I think we prefer spaces around arrays like this: `%w( production development test )...`